frontend/components/ModelForm.tsx

```diff
@@ -0,0 +1,515 @@
+                                  ? { ...this.props.defaultObject }
+                                  : {},
+                            })
+                            this.onChange({})
+                          }}
+                          className="button is-primary is-pulled-right"
+                        >
+                          {allowCreation ? (
+                            <>
+                              <Icon name="plus" alt="create" /> Create New
+                            </>
+                          ) : (
+                            <>
+                              <Icon name="x" alt="close" /> Close
+                            </>
+                          )}
+                        </span>
+                      )}
+                      <ModelStatus status={currentObject?._status} />
+                    </FormFieldClassContext.Provider>
+                  </Form>
+                )}
+              </Formik>
+            </>
+          )}
+
+          {currentObject?._error_message && (
+            <div style={{ color: 'red', marginTop: '1rem' }}>
+              <Icon name="alert-circle" />{' '}
+              {typeof currentObject._error_message === 'object' &&
+                'Errors occured while processing your request:'}
+              {formatResponse(currentObject._error_message)}
+            </div>
+          )}
+        </>
+      )
+    }
+
+    return (
+      <>
+        {objects.map((object) => (
+          <ModelItem
+            key={
+              typeof object.id === 'undefined'
+                ? `new-${object.tempId}`
+                : object.id
+            }
+          >
+            <Formik
+              initialValues={object}
+              initialStatus={
+                object && object._status === false && object._error_message
+              }
+              onSubmit={(data) => this.onSubmit(object, data)}
+              enableReinitialize
+            >
+              {({ isSubmitting }) => (
+                <Form>
+                  <FormFieldClassContext.Provider value="is-horizontal">
+                    {fields}
+                    <button
+                      type="submit"
+                      disabled={isSubmitting}
+                      className="button is-primary"
+                    >
+                      <Icon name="edit" alt="save" /> Save
+                    </button>
+                    <span
+                      className="button is-danger"
+                      style={{ marginLeft: '0.5em' }}
+                      onClick={() => this.onDelete(object)}
+                    >
+                      <Icon name="trash" alt="trash" /> Delete
+                    </span>
+                    <ModelStatus status={object._status} />
+                  </FormFieldClassContext.Provider>
+                </Form>
+              )}
+            </Formik>
+          </ModelItem>
+        ))}
+        {allowCreation && (
```

Comment:
This definitely matches the current expectation, but I know in future dashboard designs we might be splitting out the list of existing entities from the new entity form. Would it make sense to factor that out here and then make another component that puts the two components together?

Reply:
Split it out a little bit, so that the list component and model form component are separate.

It would be more difficult to split out the ModelForm into a child component, since there are actually two views that the ModelForm can be in.

One is a table view where you can edit one item at a time and the other is a inline view where you can edit all of the objects simultaneously. You can switch between the views depending on whether or not you specify the table columns.